---
title: "Untitled"
author: "yangqingxin"
date: "2024/1/8"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#1. loading package and data 
##1.1. package
```{r}
library(Seurat)
library(Matrix)
library(magrittr)
library(ggplot2)
library(data.table)
library(ggpubr)
library(cowplot)
library(readr)
library(readxl)
library(VennDiagram)
library(ComplexHeatmap)
library(viridis)
library(RobustRankAggreg)
```
##1.2. data
```{r}
#
Pbsce <- readRDS("./Analysis/Pb_IDC_seurat.Rds")
Pfsce <- readRDS("./Analysis/Pf_IDC_seurat.Rds")

#
Pbgene <- readRDS("./Analysis/Mfuzz_Pb_gene.Rds")
Pfgene <- readRDS("./Analysis/Mfuzz_Pf_gene.Rds")
plasmodium_exported_protein <- read.xlsx("./Data/Exprotein_gene_list.xlsx",sheetName = "sheet1")
```

#2. Venn diagram shows candidate genes
##2.1.
```{r}
ID_transform <- read.xlsx("./Data/Gene_Orth_Data.xlsx",sheetIndex = 1)%>%as.data.table()
setkey(ID_transform,berghei)
transID <- function(x){
  y <- ID_transform[x]$falciparum
  return(y)
}
Pbgene <- ID_transform[Pbgene]$falciparum
Pbgene <- subset(Pbgene,!is.na(Pbgene))
```
##2.2. 
###2.2.1. 
```{r}
setkey(ID_transform,falciparum)
Candidate <- Reduce(intersect,list(Pbgene,Pfgene,plasmodium_exported_protein$PF_name))
Candidate <- data.table(
  Pf= Candidate,
  Pb= ID_transform[Candidate]$berghei
)
setkey(Candidate,Pb)
```
###2.2.2. 
```{r}
A <- Pfgene
B <- Pbgene
C <- plasmodium_exported_protein$PF_name

venn.plot <- venn.diagram(
x = list(A=A,B=B,C=C),
category = c("P. falciparum","P. berghei","Exported genes"),
filename = NULL,
  fill = c( "#D6E356","#CAB2D6","#AEE6D6"), 
  lty = "blank",
  cex = 2.5,     
  cat.cex = 3,  
  # col = "transparent", 
  fontfamily = "serif", 
  fontface = "bold", 
  cat.default.pos = "text",
  cat.dist=c(.18,.18,-.12),
  cat.pos=c(8,-8,-5),
  print.mode = c("raw"),
  cat.fontfamily = "serif",scaled=F
  )
 grid.draw(venn.plot)
 system("rm VennDiagram*log")
```


#3. RRA algorithm
##3.1. Conservation of Protein sequence 
```{r}
Protein_seq <- readRDS("./Data/Protein_sequence_of_candidategenes.Rds")

seq_dist <- function(Gene){
#
x <- Protein_seq[Gene][[1]]
x <- as.character(x)
y <- protr::parSeqSim(x)
y%<>%as.data.frame()
colnames(y) <- names(x)
rownames(y) <- names(x)
y <- y[Gene,]
y %<>%melt()
y$Gene <- rep(Gene,nrow(y))
colnames(y) <- c("Target","Score","Host")

z <- data.frame(Genename = Gene,Score = mean(y$Score))
return(z)
}

#
Protein_conservation <- lapply(Candidate$Pf,seq_dist)
Protein_conservation <- do.call(rbind,Protein_conservation)%>%as.data.table()
setorder(Protein_conservation,-Score)
```

##3.2. Conservation of DNA structure
```{r}
MIS_score <- read.xlsx("./Data/Gene_MIS_Score.xlsx",sheetIndex = 1)%>%as.data.table()
setkey(MIS_score,Gene_ID)
MIS_score <- MIS_score[Candidate$Pf]
setorder(MIS_score,MIS)
```

##3.3. Conservation of gene expression 
```{r}
#
PF_expression <- AverageExpression(Pfsce,features = gsub("_","-",Candidate$Pf),group.by = "SciBet")
PF_expression <- PF_expression$RNA
rownames(PF_expression) <- gsub("-","_",rownames(PF_expression))
PB_expression <- AverageExpression(Pbsce,features = gsub("_","-",Candidate$Pb),group.by = "Cluster")
PB_expression <- PB_expression$RNA
rownames(PB_expression) <- gsub("-","_",rownames(PB_expression))
rownames(PB_expression) <- Candidate[rownames(PB_expression)]$Pf
##
identical(rownames(PF_expression),rownames(PB_expression))
identical(colnames(PF_expression),colnames(PB_expression))
#
Expression_Cor <- lapply(Candidate$Pf, function(x){
   Cor <- cor(PF_expression[x,],PB_expression[x,])
   Cor <- data.table(Gene = x, Cor=Cor)
   return(Cor)
}
                         )
Expression_Cor <- do.call(rbind,Expression_Cor)%>%as.data.table()
setorder(Expression_Cor,-Cor)
```

##3.4. Conclusion
###3.4.1. 
```{r}
glist <- list(
   a1 = Protein_conservation$Genename
  ,a2 = Expression_Cor$Gene
  ,a3_1 = MIS_score$Gene_ID
)

#应用RRA算法，对基因进行整合排序
ag <- aggregateRanks(glist)
ag$Score <- 1/ag$Score
```

###3.4.2. 
```{r,fig.height=3,fig.width=8}
#
colnames(PB_expression) <- c("ER","MR","ET","MT","LT","ES","MS","LS")
colnames(PF_expression) <- c("ER","MR","ET","MT","LT","ES","MS","LS")
pb_express <- PB_expression[ag$Name,]
pf_express <- PF_expression[ag$Name,]
pb_express%<>%t%>%scale%>%t()
pf_express%<>%t%>%scale%>%t()
#
p1 <- ComplexHeatmap::Heatmap(pf_express,
                              cluster_rows = F,cluster_columns = F,show_row_names = T,show_column_names = T,row_names_side = "left",
                              col=circlize::colorRamp2(c(-1, 0, 1), c("#2166AC", "#F7F7F7", "#B2182B")),
                              show_heatmap_legend = F,
                              column_title = "P. falciparum",
                              column_names_side = "top",column_names_rot = 45
                              )  

p2 <-ComplexHeatmap::Heatmap(pb_express,
                             cluster_rows = F,cluster_columns = F,show_row_names = T,show_column_names = T,row_names_side = "left",
                             col=circlize::colorRamp2(c(-1, 0, 1), c("#2166AC", "#F7F7F7", "#B2182B")),
                             column_title = "P. berghei",
                             column_names_side = "top",column_names_rot = 45
                             )
ha <-  HeatmapAnnotation(barplot = anno_barplot(ag$Score, 
                                                which = "row", 
                                                bar_width=0.6, 
                                                gp= gpar(fill=c("red","black","black","black","black","black"))),
                         which = "row")

p1+(add_heatmap(p2, ha))
```
